File: spring-boot-iot-ui/src/api/region.ts
import { request } from './request'
import type { ApiEnvelope, IdType, PageResult } from '../types/api'

export interface Region {
  id: IdType
  tenantId: IdType
  regionName: string
  regionCode: string
  parentId: IdType
  regionType: string
  longitude?: number
  latitude?: number
  status: number
  sortNo: number
  remark: string
  createBy: number
  createTime: string
  updateBy: number
  updateTime: string
  deleted: number
  children?: Region[]
  hasChildren?: boolean
}

export interface RegionQueryParams {
  regionName?: string
  regionCode?: string
  regionType?: string
}

export interface RegionPageQueryParams extends RegionQueryParams {
  pageNum?: number
  pageSize?: number
}

function buildQuery(params: Record<string, unknown>) {
  const query = new URLSearchParams()
  Object.entries(params).forEach(([key, value]) => {
    if (value !== undefined && value !== null && value !== '') {
      query.append(key, String(value))
    }
  })
  return query.toString()
}

export const listRegions = (parentId?: IdType): Promise<ApiEnvelope<Region[]>> => {
  const query = buildQuery({ parentId })
  return request<Region[]>(`/api/region/list${query ? `?${query}` : ''}`, { method: 'GET' })
}

export const pageRegions = (params: RegionPageQueryParams = {}): Promise<ApiEnvelope<PageResult<Region>>> => {
  const query = buildQuery(params)
  return request<PageResult<Region>>(`/api/region/page${query ? `?${query}` : ''}`, { method: 'GET' })
}

export const listRegionTree = (): Promise<ApiEnvelope<Region[]>> => {
  return request<Region[]>('/api/region/tree', { method: 'GET' })
}

export const getRegion = (id: IdType): Promise<ApiEnvelope<Region>> => {
  return request<Region>(`/api/region/${id}`, { method: 'GET' })
}

export const addRegion = (data: Partial<Region>): Promise<ApiEnvelope<Region>> => {
  return request<Region>('/api/region', { method: 'POST', body: data })
}

export const updateRegion = (data: Partial<Region>): Promise<ApiEnvelope<Region>> => {
  return request<Region>('/api/region', { method: 'PUT', body: data })
}

export const deleteRegion = (id: IdType): Promise<ApiEnvelope<void>> => {
  return request<void>(`/api/region/${id}`, { method: 'DELETE' })
}

File: spring-boot-iot-system/src/main/java/com/ghlzm/iot/system/entity/Region.java
package com.ghlzm.iot.system.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;
import java.util.List;

/**
 * 区域管理实体
 */
@Data
@TableName("sys_region")
public class Region implements Serializable {
      private static final long serialVersionUID = 1L;

      /**
       * 主键
       */
      @TableId(value = "id", type = IdType.ASSIGN_ID)
      private Long id;

      /**
       * 租户ID
       */
      private Long tenantId;

      /**
       * 区域名称
       */
      private String regionName;

      /**
       * 区域编码
       */
      private String regionCode;

      /**
       * 父ID
       */
      private Long parentId;

      /**
       * 区域类型 province/city/district/street
       */
      private String regionType;

      /**
       * 经度
       */
      private BigDecimal longitude;

      /**
       * 纬度
       */
      private BigDecimal latitude;

      /**
       * 状态 1启用 0禁用
       */
      private Integer status;

      /**
       * 排序
       */
      private Integer sortNo;

      /**
       * 备注
       */
      private String remark;

      /**
       * 创建人
       */
      private Long createBy;

      /**
       * 创建时间
       */
      private Date createTime;

      /**
       * 更新人
       */
      private Long updateBy;

      /**
       * 更新时间
       */
      private Date updateTime;

      /**
       * 删除标记
       */
      @TableLogic
      private Integer deleted;

      /**
       * 子节点
       */
      @TableField(exist = false)
      private List<Region> children;

      @TableField(exist = false)
      private Boolean hasChildren;
}
